janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/AbstractIndexStatusWatcher.java

```diff
@@ -22,34 +22,40 @@
 import java.time.Duration;
 import java.time.temporal.TemporalUnit;
 import java.util.concurrent.Callable;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
 
 public abstract class AbstractIndexStatusWatcher<R, S extends AbstractIndexStatusWatcher<R,S>> implements Callable<R> {
 
     protected JanusGraph g;
-    protected SchemaStatus status;
+    protected ArrayList<SchemaStatus> statuses = new ArrayList<SchemaStatus>();
```

Comment:
things werent final in this class before; perhaps there should be a different WI open to go through the entire project and make everything `final`?